### DISCUSSION TITLE
[ Unstable ] Multiplexer doesn't respond to 'set_input' 'move_input' commands.

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#14408] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The 'Demultiplexer' works as intended, however the 'Multiplexer' aka "Input Selector" refuses to change it's signal even when using the same working input setup with outputs switched around, or even trying to force it to change with a value in 'set_input'.

![image](https://github.com/user-attachments/assets/5fecd931-b999-4fc9-a0a5-6cd2c16ca93c)
![image](https://github.com/user-attachments/assets/8bc3ba04-3b09-440c-8c3c-b586588503e8)

### Reproduction steps

- Fill multiplexer inputs with signals & connect multiplexer output.
- Wire any sort of buttons or console connections with (1) and (-1) to 'move_input' same as the camera demultiplexer prefab. Or try to change the input directly with some other value in 'set_input'
- The multiplexer will refuse to change what input is selected.

however accessing the multiplexer with your character and manually changing the selected input does update the signal correctly, so its just the switching mechanic that seems to be broken.
![image](https://github.com/user-attachments/assets/ad3b1920-4fbd-4342-bca7-d066ebb0f571)

Same results in subeditor test mode, quickstart, or even in campaign.
Haven't tested in multiplayer.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.101.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
Partial duplicate of https://github.com/FakeFishGames/Barotrauma/discussions/14345#discussion-6986268 although that report covers other issues with circuit boxes & didn't see it at first because github discussion search function sucks I guess.

**Comment by Regalis11** [comment]
I believe this has been addressed as a part of the fixes to address https://github.com/FakeFishGames/Barotrauma/discussions/14345. The fixes are coming up in the next unstable build, sorry for the inconvenience!
